src/meta/src/hummock/manager/mod.rs

```diff
@@ -2459,6 +2404,235 @@ where
             }
         }
     }
+
+    pub async fn compaction_event_loop(
+        hummock_manager: Arc<Self>,
+        mut compactor_streams_change_rx: UnboundedReceiver<(
+            u32,
+            Streaming<SubscribeCompactionEventRequest>,
+        )>,
+    ) -> (JoinHandle<()>, Sender<()>) {
+        use futures::StreamExt;
+        let mut compactor_request_streams = FuturesUnordered::new();
+        let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+        let shutdown_rx_shared = shutdown_rx.shared();
+        let mut compaction_selectors = init_selectors();
+
+        let join_handle = tokio::spawn(async move {
+            let push_stream =
+                |context_id: u32,
+                 stream: Streaming<SubscribeCompactionEventRequest>,
+                 compactor_request_streams: &mut FuturesUnordered<_>| {
+                    let future = stream
+                        .into_future()
+                        .map(move |stream_future| (context_id, stream_future));
+
+                    compactor_request_streams.push(future);
+                };
+
+            loop {
+                let shutdown_rx_shared = shutdown_rx_shared.clone();
+
+                tokio::select! {
```

Comment:
Do we need this select to be biased?

Reply:
When we have `pending_on_none`, priority doesn't matter.  cc@wenym1